### DISCUSSION TITLE
Textadept Windows support history

### DISCUSSION THREAD

**Discussion by LifeCANvs** [orbitalquark/textadept#542] [General]
Hi, I am helping some kids learn about (a) history of computing, including the evolution of Operating Systems and (b) programming. So I am letting them work on some programming projects on older Windows systems, including Windows 98SE, 95 and even Windows 3.1 with and without win32s, to experience the differences in usability, implementations, restrictions, etc. on the various historic systems. I have not managed to find a relatively simple code editor that runs on modern systems and also on these legacy/"ancient" systems... except when I discovered that Textadept seems to have come along from quite far in the computing history... so I am hoping that it did at some stage in the past have versions running on those old Windows OSs.

Would it be possible to run old versions of Textadept on (1) **Windows 98SE**, (2) Windows **95** SR2.5, (3) Windows **3.1 with win32s** (I assume Ta will not work on Windows 3.1 standard 16-bit without the 32-bit extensions, as Textadept only ever supported win32 and not win16)? If so, could someone please tell me which were the latest versions of Textadept to still support (any of) those respective systems / platforms?

I see this as a very useful and fun learning experience for all involved in the project with me, so I would really appreciate any feedback that can help me to make it happen for the kids. And I am also enjoying the challenge of "traveling back in time" and revisiting the experiences of using these old systems. I'm actually wondering whether a past TUI version of Textadept might be able to run on FreeDOS and possibly on MS-**DOS with a 32-bit DPMI extender** such as DOS4GW / HX DOS Extender?

I went through all of the Textadept history / changelog resources that I could find on the Web and in the Internet Archive Wayback machine, but the only clear indications of Windows OS support versions are for when Windows XP support ended (up to Textadept 11.3), and then more modern Windows systems versions support (for Win 7, 10...). It seems that Lua supported Windows NT/2000 up to v 5.3.4, but no longer since v 5.3.5. From the Textadept changelogs I learned that you did use Lua 5.3 and older, so that should theoretically mean that it could have supported back to at least Windows NT 4 and possibly NT 3.5 (and maybe NT 3.1), which could also mean there is a chance that it could run on Windows 3.1 with win32s.

Whether Textadept ever worked on non-NT Windows systems such as Windows 9x, is of course a different matter, but I would guess that Textadept probably did at some stage work on at least Windows 98SE and newer, as those Windwos systems were very widely used, and also because I did discover that Scintilla supported Windows 9x up to version 3.5.2, which was also used for Textadept in the past, which gives me hope that Textadept also worked on Windows 98SE and 95.

Any info regarding the above questions would be greatly appreciated.

Kind regards, and thanks for all the hard work on Textadept through so many years...

**Comment by orbitalquark** [comment]
That sounds like a fun project! Unfortunately, Windows XP was the earliest supported version of Windows. Earlier versions of Textadept used the GTK toolkit (2.x), which I believe only ever supported Windows XP.

You might want to try SciTE (https://www.scintilla.org/SciTE.html), which may go back further than Windows XP. (Textadept and SciTE share the same editing component, Scintilla.)
